File: doccoon_backend/doccoon/serializers/ai.py
from rest_framework import serializers


class AIRefineRequestSerializer(serializers.Serializer):
    content = serializers.CharField()
    mode = serializers.ChoiceField(choices=["refine", "rewrite"], default="refine")
    context = serializers.CharField(required=False, default="", allow_blank=True)


class AIRefineResponseSerializer(serializers.Serializer):
    original = serializers.CharField()
    refined = serializers.CharField()
    mode = serializers.CharField()

File: doccoon_frontend/src/api/ai.ts
import { api } from "@/api/client";

export interface RefineRequest {
  content: string;
  mode: "refine" | "rewrite";
  context?: string;
}

export interface RefineResponse {
  original: string;
  refined: string;
  mode: string;
}

export async function refineContent(
  payload: RefineRequest,
): Promise<RefineResponse | null> {
  const response = await api.post<RefineResponse>("/ai/refine/", payload);
  return response.results ?? null;
}
